subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/KotlinDslAssignmentIntegrationTest.kt

```diff
@@ -181,6 +181,63 @@ class KotlinDslAssignmentIntegrationTest : AbstractKotlinIntegrationTest() {
         build("myTask")
     }
 
+    @Test
+    fun `assign operator compiles with all possible Property types`() {
+        // Given
+        withAssignmentOverload()
+        withBuildScript("""
+            abstract class MyTask : DefaultTask() {
+                @get:Input
+                abstract val input: Property<String>
+                @get:Input
+                abstract val mapInput: MapProperty<String, String>
+                @get:Input
+                abstract val setInput: SetProperty<String>
+                @get:Input
+                abstract val listInput: ListProperty<String>
+                @get:InputDirectory
+                abstract val dirInput: DirectoryProperty
+                @get:OutputFile
+                abstract val fileOutput: RegularFileProperty
+
+                @TaskAction
+                fun taskAction() {
+                    fileOutput.asFile.get().writeText(input.get())
+                }
+            }
+
+            tasks.register<MyTask>("myTask") {
+                file("src").mkdirs()
+                input = null
+                input = "Hello world"
+                input = provider { null }
+                input = provider { "Hello" }
+                mapInput = null
+                mapInput = mapOf<String, String>("a" to "b")
```

Comment:
:question: isn't the type argument unnecessary here?
```suggestion
                mapInput = mapOf("a" to "b")
```
Same for other statements below

Reply:
Yes,  this can be simplified. Good catch